traits/ctraits.c

```diff
@@ -3908,7 +3919,7 @@ validate_trait_complex(
             case 9: /* Tuple item check: */
                 result = validate_trait_tuple_check(
                     PyTuple_GET_ITEM(type_info, 1), obj, name, value);
-                if (result != NULL) {
+                if (result != NULL || PyErr_Occurred()) {
```

Comment:
The `||` path here is not currently exercised by the regression test. Fixing ...

Reply:
Fixed.